### PR TITLE
feat(worksession): add work-session command

### PR DESCRIPTION
**Pull request by jisung-02**
## Summary

- Add \`alpacon work-session\` (alias: \`session\`) command group with six subcommands: \`list\`, \`create\`, \`describe\`, \`activate\`, \`complete\`, \`extend\`
- Work sessions are approval-gated units that group Websh, WebFTP, and exec access on Alpacon
- Support both interactive and non-interactive modes across all subcommands

## Subcommands

| Command | Description |
|---------|-------------|
| \`list\` | List work sessions with optional \`--status\` and \`--requester-type\` filters |
| \`create\` | Create a new work session; \`--wait\` polls until approved |
| \`describe\` | Show full details of a work session |
| \`activate\` | Activate an approved work session |
| \`complete\` | Mark an active work session as completed |
| \`extend\` | Extend the expiry of an approved or active work session |

## Key design decisions

- **Interactive/non-interactive support**: missing required flags prompt in TTY; error-exit in non-interactive mode (\`IsInteractiveShell\` guard before every \`PromptForRequiredInput\` call)
- **\`--expires-in\` / \`--expires-at\` mutual exclusion**: accepts a duration (e.g. \`2h\`) or an RFC3339 absolute timestamp, never both
- **\`--wait\` flag**: polls every 10 seconds up to 30 attempts after \`create\`, then exits on approval or rejection
- **Agent scope restriction**: \`websh\` scope is disallowed when \`--requester-type agent\`
- **Server name → ID resolution**: \`--servers\` accepts human-readable names; the CLI resolves them to IDs before sending the request

## Test plan

- [x] \`api/worksession\`: full API layer coverage via \`httptest.NewServer\` (list, create, get, activate, complete, extend)
- [x] \`cmd/worksession\`: unit tests for \`parseExpiryFlag\`, \`validateAgentScopes\`, \`splitCSV\`
- [x] \`go test -race ./cmd/worksession/... ./api/worksession/...\` passes
- [x] Manually tested all subcommands against a live Alpacon environment: \`ls\`, \`create\`, \`describe\`, \`extend\`, \`complete\`, \`activate\`

### CODE REVIEW

File: cmd/worksession/worksession_list.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"github.com/alpacax/alpacon-cli/client"
+	"github.com/alpacax/alpacon-cli/utils"
+	"github.com/spf13/cobra"
+)
+
+var workSessionListCmd = &cobra.Command{
+	Use:     "list",
+	Aliases: []string{"ls"},
+	Short:   "List work sessions",
+	Example: `  alpacon work-session list
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_list.go]: Fixed. Changed `Use: "list"` with `Aliases: ["ls"]` to `Use: "ls"` with `Aliases: ["list"]` to align with the existing convention used by `server_list.go`, `user_list.go`, and other list-style commands in the repo. Updated examples accordingly. (commit 5152dfb)

File: cmd/worksession/worksession_create.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		if err != nil {
+			return fmt.Errorf("polling failed: %w", err)
+		}
+		switch s.Status {
+		case "approved":
+			return nil
+		case "rejected":
+			return errors.New("work session was rejected")
+		}
+		utils.CliInfo("Waiting for approval... (attempt %d/%d)", attempt, maxAttempts)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:200]: Fixed. Added `expired`, `revoked`, and `completed` as terminal status cases in `pollForApproval` so the CLI exits immediately with a descriptive error instead of polling until timeout. Verified against the server-side `STATUS_CHOICES` definition—these are all the non-pending, non-approved statuses defined on the model. (commit 1046cd7)

File: cmd/worksession/worksession_create.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	parts := strings.Split(s, ",")
+	result := make([]string, 0, len(parts))
+	for _, p := range parts {
+		p = strings.TrimSpace(p)
+		if p != "" {
+			result = append(result, p)
+		}
+	}
+	return result
+}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:175]: Fixed. Added `TestSplitCSV` with 7 table-driven test cases covering normal input, whitespace trimming, trailing/leading commas, empty input, single value, and commas-only edge cases. All tests pass. (commit 5d09590)

File: cmd/worksession/worksession_list.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package worksession
+
+import (
+	"github.com/alpacax/alpacon-cli/api/worksession"
+	"github.com/alpacax/alpacon-cli/client"
+	"github.com/alpacax/alpacon-cli/utils"
+	"github.com/spf13/cobra"
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_list.go:7]: Fixed. The import is now aliased as `wsapi` (consistent with `worksession_create.go` and `worksession_describe.go`).

File: cmd/worksession/worksession_list.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		}
+
+		utils.PrintTable(sessions)
+	},
+}
+
+func init() {
+	workSessionListCmd.Flags().StringVar(&statusFilter, "status", "", "Filter by status (pending, approved, active, completed, rejected, expired, revoked)")
+	workSessionListCmd.Flags().StringVar(&requesterFilter, "requester-type", "", "Filter by requester type (user, agent)")
+}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_list.go:39]: Fixed. Added client-side validation before the API call — only `"user"` and `"agent"` are accepted; any other value exits early with a clear error message.

File: cmd/worksession/worksession_create.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+				}
+			} else {
+				utils.CliErrorWithExit("Invalid expiry: %s.", err)
+			}
+		}
+
+		scopeList := splitCSV(scopes)
+		if err := validateAgentScopes(requesterType, scopeList); err != nil {
+			utils.CliErrorWithExit("Invalid --scopes: %s", err)
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:81]: Fixed. Added an explicit check right before scope validation — unsupported values are rejected early with a user-friendly error message.

File: cmd/worksession/worksession_create.go [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+			utils.CliErrorWithExit("Connection to Alpacon API failed: %s. Consider re-logging.", err)
+		}
+
+		serverList := splitCSV(servers)
+		serverIDs := make([]string, 0, len(serverList))
+		for _, name := range serverList {
+			id, err := server.GetServerIDByName(ac, name)
+			if err != nil {
+				utils.CliErrorWithExit("Server %q not found: %s.", name, err)
+			}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:97]: Acknowledged. `GetServerIDByName` issues a filtered single-item list request per name, so the cost is proportional to the number of servers specified. In practice, work sessions target a small, explicit set of servers (typically 1–3), making the N+1 impact negligible. Batch name-to-ID resolution would require a new API endpoint; deferring to a follow-up if it becomes a bottleneck.

File: cmd/worksession/worksession_create.go [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+		return "", errors.New("--expires-in and --expires-at are mutually exclusive")
+	}
+	if expiresIn == "" && expiresAt == "" {
+		return "", errors.New("one of --expires-in or --expires-at is required")
+	}
+	if expiresIn != "" {
+		d, err := time.ParseDuration(expiresIn)
+		if err != nil {
+			return "", fmt.Errorf("invalid --expires-in value %q: %w", expiresIn, err)
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:140]: Fixed. Added an explicit check after parsing the duration: if `d <= 0`, `parseExpiryFlag` returns an error requiring a positive duration. Tests for zero (`0s`) and negative (`-1h`) cases are included.

File: cmd/worksession/worksession_create.go [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+	for attempt := 1; attempt <= maxAttempts; attempt++ {
+		s, err := wsapi.GetWorkSession(ac, id)
+		if err != nil {
+			return fmt.Errorf("polling failed: %w", err)
+		}
+		switch s.Status {
+		case "approved":
+			return nil
+		case "rejected":
+			return errors.New("work session was rejected")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:192]: Fixed. Added `"active"` as a success terminal state alongside `"approved"`. Since `active` is a post-approval state (someone approved and immediately activated), treating it as success correctly handles the race between the poll interval and a rapid approval+activation.

File: api/worksession/worksession.go [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+	if err = json.Unmarshal(body, &session); err != nil {
+		return nil, err
+	}
+	return &session, nil
+}
+
+func ActivateWorkSession(ac *client.AlpaconClient, id string) error {
+	url := fmt.Sprintf("%s%s/activate/", workSessionURL, id)
+	_, err := ac.SendPostRequest(url, struct{}{})
+	return err
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/worksession/worksession.go:73]: Fixed. Replaced `fmt.Sprintf` with `utils.BuildURL(workSessionURL, path.Join(id, "activate"), nil)`, consistent with `websh.go` and other API packages.

File: api/worksession/worksession.go [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+func CompleteWorkSession(ac *client.AlpaconClient, id string) error {
+	url := fmt.Sprintf("%s%s/complete/", workSessionURL, id)
+	_, err := ac.SendPostRequest(url, struct{}{})
+	return err
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/worksession/worksession.go:78]: Fixed. Same pattern applied: `utils.BuildURL(workSessionURL, path.Join(id, "complete"), nil)`.

File: api/worksession/worksession.go [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+func ExtendWorkSession(ac *client.AlpaconClient, id string, req WorkSessionExtendRequest) error {
+	url := fmt.Sprintf("%s%s/extend/", workSessionURL, id)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/worksession/worksession.go]: Fixed. Same pattern applied: `utils.BuildURL(workSessionURL, path.Join(id, "extend"), nil)`.

File: cmd/worksession/worksession_create.go [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+		case "expired":
+			return errors.New("work session expired while waiting for approval")
+		case "revoked":
+			return errors.New("work session was revoked")
+		case "completed":
+			return errors.New("work session was completed unexpectedly")
+		}
+		utils.CliInfo("Waiting for approval... (attempt %d/%d)", attempt, maxAttempts)
+		time.Sleep(interval)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go]: Fixed in 192fd6c. Added a guard so `time.Sleep` is only called when `attempt < maxAttempts`, eliminating the unnecessary 10-second delay before the timeout error is returned.

File: cmd/worksession/worksession.go [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+var WorkSessionCmd = &cobra.Command{
+	Use:     "work-session",
+	Aliases: []string{"session"},
+	Short:   "Create and manage work sessions",
+	Long:    "Create, inspect, and manage work sessions — the approval-gated units that group Websh, WebFTP, and exec access on Alpacon.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := cmd.Help(); err != nil {
+			return err
+		}
+		return errors.New("a subcommand is required. Run 'alpacon work-session --help' for more information")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession.go]: Fixed in 192fd6c. Expanded the error message to list all available subcommands (`ls`, `create`, `describe`, `activate`, `complete`, `extend`), matching the pattern used by `server` and `user` command groups.

File: cmd/worksession/worksession_create.go [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+func init() {
+	workSessionCreateCmd.Flags().StringVar(&purpose, "purpose", "", "Session purpose")
+	workSessionCreateCmd.Flags().StringVar(&scopes, "scopes", "", "Comma-separated scopes (command, websh, webftp, editor, tunnel, sudo)")
+	workSessionCreateCmd.Flags().StringVar(&servers, "servers", "", "Comma-separated server names")
+	workSessionCreateCmd.Flags().StringVar(&expiresIn, "expires-in", "", "Session duration (e.g. 1h, 2h, 4h)")
+	workSessionCreateCmd.Flags().StringVar(&expiresAt, "expires-at", "", "Absolute expiry time (RFC3339)")
+	workSessionCreateCmd.Flags().StringVar(&requesterType, "requester-type", "user", "Requester type: user or agent")
+	workSessionCreateCmd.Flags().BoolVar(&waitApproval, "wait", false, "Poll until the session is approved, then exit")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:215]: Fixed in b9854d6. Added `workSessionCreateCmd.MarkFlagsMutuallyExclusive("expires-in", "expires-at")` to the `init()` function, matching the pattern from `server_create.go`. The runtime `parseExpiryFlag` check is retained for the required-one-of validation and interactive prompting.

File: cmd/worksession/worksession_extend.go [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+			utils.CliErrorWithExit("Failed to extend work session: %s.", err)
+		}
+
+		utils.CliSuccess("Work session %s extended to %s.", args[0], expiresAtVal)
+	},
+}
+
+func init() {
+	workSessionExtendCmd.Flags().StringVar(&extendExpiresIn, "expires-in", "", "New expiry set to now + duration (e.g. 2h)")
+	workSessionExtendCmd.Flags().StringVar(&extendExpiresAt, "expires-at", "", "New absolute expiry time (RFC3339)")
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_extend.go:54]: Fixed in b9854d6. Added `workSessionExtendCmd.MarkFlagsMutuallyExclusive("expires-in", "expires-at")` to the `init()` function, consistent with the create command and the pattern from `server_create.go`.

File: cmd/worksession/worksession_create.go [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+				utils.CliErrorWithExit("Non-interactive mode requires --scopes.")
+			}
+			scopes = utils.PromptForRequiredInput("Scopes (comma-separated, e.g. command,websh): ")
+		}
+		if servers == "" {
+			if !utils.IsInteractiveShell() {
+				utils.CliErrorWithExit("Non-interactive mode requires --servers.")
+			}
+			servers = utils.PromptForRequiredInput("Servers (comma-separated server names): ")
+		}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:53]: Fixed in 451ae7a. Added `strings.TrimSpace` to `purpose`, `scopes`, and `servers` before the empty-string check, so whitespace-only flag values (e.g. `--purpose "  "`) are treated as missing and trigger the interactive prompt or non-interactive error accordingly.

File: cmd/worksession/worksession_create.go [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+
+		ac, err := client.NewAlpaconAPIClient()
+		if err != nil {
+			utils.CliErrorWithExit("Connection to Alpacon API failed: %s. Consider re-logging.", err)
+		}
+
+		serverList := splitCSV(servers)
+		serverIDs := make([]string, 0, len(serverList))
+		for _, name := range serverList {
+			id, err := server.GetServerIDByName(ac, name)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:94]: Fixed in 451ae7a. Added `len(scopeList) == 0` and `len(serverList) == 0` guards after `splitCSV` to catch inputs like `",,"` or `" , "` that produce an empty slice, preventing an empty `Scopes`/`Servers` list from being sent to the API.

File: cmd/worksession/worksession_create.go [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+		}
+		if d <= 0 {
+			return "", fmt.Errorf("invalid --expires-in value %q: must be a positive duration", expiresIn)
+		}
+		return time.Now().UTC().Add(d).Format(time.RFC3339), nil
+	}
+	if _, err := time.Parse(time.RFC3339, expiresAt); err != nil {
+		return "", fmt.Errorf("invalid --expires-at value %q: must be RFC3339 format", expiresAt)
+	}
+	return expiresAt, nil
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:149]: Fixed in 451ae7a. Added `strings.TrimSpace` to both `expiresIn` and `expiresAt` at the start of `parseExpiryFlag`, so values with leading/trailing whitespace (e.g. `" 2h "`) are parsed correctly and the mutual-exclusion/required checks remain accurate after trimming.

File: cmd/worksession/worksession.go [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+var (
+	statusFilter    string
+	requesterFilter string
+)
+
+var WorkSessionCmd = &cobra.Command{
+	Use:     "work-session",
+	Aliases: []string{"session"},
+	Short:   "Create and manage work sessions",
+	Long:    "Create, inspect, and manage work sessions — the approval-gated units that group Websh, WebFTP, and exec access on Alpacon.",
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession.go]: Fixed. Removed the surrounding spaces around the em dash to match the repository's writing convention.

File: cmd/worksession/worksession_create.go [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+		return errors.New("\"websh\" is not allowed for agent sessions")
+	}
+	return nil
+}
+
+// splitCSV splits a comma-separated string and trims whitespace.
+func splitCSV(s string) []string {
+	parts := strings.Split(s, ",")
+	result := make([]string, 0, len(parts))
+	for _, p := range parts {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go:168]: Acknowledged. Deduplicating `splitCSV` and `splitAndTrim` into a shared `utils` helper is a valid improvement, but it touches files outside this PR's scope. I'll address it in a follow-up refactoring PR to keep this change focused.

File: cmd/worksession/worksession_create.go [hunk 22]
```diff
@@ -0,0 +1,217 @@
...
+		p = strings.TrimSpace(p)
+		if p != "" {
+			result = append(result, p)
+		}
+	}
+	return result
+}
+
+// pollForApproval polls GET /work-sessions/sessions/{id}/ every 10 seconds
+// until status is "approved" or 30 attempts are exhausted.
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/worksession/worksession_create.go]: Fixed. Updated the comment to accurately reflect all handled statuses (approved/active as success, rejected/expired/revoked/completed as early exits).

**Review by eunyoung14 — APPROVED** [review]
